components/sbm-recipes-mule-to-boot/src/main/java/org/springframework/sbm/mule/actions/JavaDSLAction2.java

```diff
@@ -96,12 +96,12 @@ private void handleTopLevelElements(BuildFile buildFile, MuleMigrationContext mu
                 topLevelElements.add(new UnknownTopLevelElement(tle));
             }
         }
-        List<Dependency> listOfDependencies = topLevelElements.stream()
+        Set<Dependency> dependencies = topLevelElements.stream()
                 .map(this::buildDependencies)
                 .flatMap(List::stream)
-                .collect(Collectors.toList());
-        logEvent("Adding " + listOfDependencies.size() + " dependencies");
-        buildFile.addDependencies(listOfDependencies);
+                .collect(Collectors.toSet());
```

Comment:
Here a `startProcess()` and `endProcess()` around `buildFile.addDependencies(new ArrayList<>(dependencies))` would maybe be better as this step takes quite a while and for a process a loader is displayed.
`logEvent()` would be used to log something that happened (event).

Reply:
done